src/miner.cpp

```diff
@@ -435,6 +421,36 @@ static bool ProcessBlockFound(CBlock* pblock, CWallet& wallet, CReserveKey& rese
     return true;
 }
 
+bool MineBlock(CReserveKey& reservekey, uint256& hash)
+{
+    unsigned int nExtraNonce = 0;
+
+    while (true) {
+        CBlockIndex *pindexPrev = NULL;
+
+        auto_ptr<CBlockTemplate> pblocktemplate(CreateNewBlockWithKey(reservekey, pindexPrev));
+        if (!pblocktemplate.get()) {
+            return false;
+        }
+
+        CBlock *pblock = &pblocktemplate->block;
+        IncrementExtraNonce(pblock, pindexPrev, nExtraNonce);
+
+        while (true) {
+            if (ScanHash(pblock, pindexPrev, 0x1000)) {
+                CValidationState state;
+                if (ProcessNewBlock(state, NULL, pblock)) {
+                    hash = pblock->GetHash();
```

Comment:
CreateNewBlock builds a new CBlock object. If we'd want to return it in a passed pointer, you'd have to copy it there.

Alternatively, we could return a CBlock* variable in a CBlock*& argument, but why bother? All we need is the hash...